app/models/publisher.rb

```diff
@@ -19,6 +19,27 @@ def publish
     end
   end
 
+  def discard_draft
+    begin
+      ActiveRecord::Base.transaction do
+        publishing_api.discard_draft(content_model.content_id)
+
+        latest_published_edition = content_model.editions.published.last
+        if latest_published_edition.present?
+          content_model
+            .editions
+            .where("created_at > ?", latest_published_edition.created_at)
+            .destroy_all
```

Comment:
I don't think I understand. I'm talking about refactoring the existing functionality. The SQL should be in the model.